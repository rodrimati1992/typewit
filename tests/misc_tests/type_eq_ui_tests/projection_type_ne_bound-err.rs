use typewit::{TypeNe, TypeFn};

struct Mapper;

impl<T> TypeFn<T> for Mapper {
    type Output = [T; 1];
}

fn map_to_different_type<'a, L, R>(te: TypeNe<L, R>) -> TypeNe<[L; 1], [R; 1]> {
    te.map(Mapper)
}
fn project_to_different_type<'a, L, R>(te: TypeNe<L, R>) -> TypeNe<[L; 1], [R; 1]> {
    te.project::<Mapper>()
}


fn main() {}