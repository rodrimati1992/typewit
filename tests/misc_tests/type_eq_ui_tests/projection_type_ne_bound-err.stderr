error[E0277]: the trait bound `Mapper: RevTypeFn<[L; 1]>` is not satisfied
  --> tests/misc_tests/type_eq_ui_tests/projection_type_ne_bound-err.rs:10:12
   |
10 |     te.map(Mapper)
   |        --- ^^^^^^ the trait `RevTypeFn<[L; 1]>` is not implemented for `Mapper`
   |        |
   |        required by a bound introduced by this call
   |
   = help: the following other types implement trait `RevTypeFn<Ret>`:
             <FnRev<F> as RevTypeFn<R>>
             <GRef<'a> as RevTypeFn<&'a T>>
             <GRefMut<'a> as RevTypeFn<&'a mut T>>
             <GBox as RevTypeFn<Box<T>>>
             <FnIdentity as RevTypeFn<T>>
             <Invoke<F> as RevTypeFn<R>>
             <PhantomData<F> as RevTypeFn<R>>
   = note: required for `Mapper` to implement `InjTypeFn<L>`
note: required by a bound in `typewit::type_ne_::<impl TypeNe<L, R>>::map`
  --> src/./type_ne/extra_type_ne_methods.rs
   |
   |     pub const fn map<F>(
   |                  --- required by a bound in this associated function
...
   |         InvokeAlias<F>: InjTypeFn<L> + InjTypeFn<R>
   |                         ^^^^^^^^^^^^ required by this bound in `typewit::type_ne_::<impl TypeNe<L, R>>::map`

error[E0277]: the trait bound `Mapper: RevTypeFn<[R; 1]>` is not satisfied
  --> tests/misc_tests/type_eq_ui_tests/projection_type_ne_bound-err.rs:10:12
   |
10 |     te.map(Mapper)
   |        --- ^^^^^^ the trait `RevTypeFn<[R; 1]>` is not implemented for `Mapper`
   |        |
   |        required by a bound introduced by this call
   |
   = help: the following other types implement trait `RevTypeFn<Ret>`:
             <FnRev<F> as RevTypeFn<R>>
             <GRef<'a> as RevTypeFn<&'a T>>
             <GRefMut<'a> as RevTypeFn<&'a mut T>>
             <GBox as RevTypeFn<Box<T>>>
             <FnIdentity as RevTypeFn<T>>
             <Invoke<F> as RevTypeFn<R>>
             <PhantomData<F> as RevTypeFn<R>>
   = note: required for `Mapper` to implement `InjTypeFn<R>`
note: required by a bound in `typewit::type_ne_::<impl TypeNe<L, R>>::map`
  --> src/./type_ne/extra_type_ne_methods.rs
   |
   |     pub const fn map<F>(
   |                  --- required by a bound in this associated function
...
   |         InvokeAlias<F>: InjTypeFn<L> + InjTypeFn<R>
   |                                        ^^^^^^^^^^^^ required by this bound in `typewit::type_ne_::<impl TypeNe<L, R>>::map`

error[E0277]: the trait bound `Mapper: RevTypeFn<[L; 1]>` is not satisfied
  --> tests/misc_tests/type_eq_ui_tests/projection_type_ne_bound-err.rs:13:18
   |
13 |     te.project::<Mapper>()
   |        -------   ^^^^^^ the trait `RevTypeFn<[L; 1]>` is not implemented for `Mapper`
   |        |
   |        required by a bound introduced by this call
   |
   = help: the following other types implement trait `RevTypeFn<Ret>`:
             <FnRev<F> as RevTypeFn<R>>
             <GRef<'a> as RevTypeFn<&'a T>>
             <GRefMut<'a> as RevTypeFn<&'a mut T>>
             <GBox as RevTypeFn<Box<T>>>
             <FnIdentity as RevTypeFn<T>>
             <Invoke<F> as RevTypeFn<R>>
             <PhantomData<F> as RevTypeFn<R>>
   = note: required for `Mapper` to implement `InjTypeFn<L>`
note: required by a bound in `typewit::type_ne_::<impl TypeNe<L, R>>::project`
  --> src/./type_ne/extra_type_ne_methods.rs
   |
   |     pub const fn project<F>(
   |                  ------- required by a bound in this associated function
...
   |         InvokeAlias<F>: InjTypeFn<L> + InjTypeFn<R>
   |                         ^^^^^^^^^^^^ required by this bound in `typewit::type_ne_::<impl TypeNe<L, R>>::project`

error[E0277]: the trait bound `Mapper: RevTypeFn<[R; 1]>` is not satisfied
  --> tests/misc_tests/type_eq_ui_tests/projection_type_ne_bound-err.rs:13:18
   |
13 |     te.project::<Mapper>()
   |        -------   ^^^^^^ the trait `RevTypeFn<[R; 1]>` is not implemented for `Mapper`
   |        |
   |        required by a bound introduced by this call
   |
   = help: the following other types implement trait `RevTypeFn<Ret>`:
             <FnRev<F> as RevTypeFn<R>>
             <GRef<'a> as RevTypeFn<&'a T>>
             <GRefMut<'a> as RevTypeFn<&'a mut T>>
             <GBox as RevTypeFn<Box<T>>>
             <FnIdentity as RevTypeFn<T>>
             <Invoke<F> as RevTypeFn<R>>
             <PhantomData<F> as RevTypeFn<R>>
   = note: required for `Mapper` to implement `InjTypeFn<R>`
note: required by a bound in `typewit::type_ne_::<impl TypeNe<L, R>>::project`
  --> src/./type_ne/extra_type_ne_methods.rs
   |
   |     pub const fn project<F>(
   |                  ------- required by a bound in this associated function
...
   |         InvokeAlias<F>: InjTypeFn<L> + InjTypeFn<R>
   |                                        ^^^^^^^^^^^^ required by this bound in `typewit::type_ne_::<impl TypeNe<L, R>>::project`
